src/components/SelectChallenge.tsx

```diff
@@ -34,20 +30,24 @@ export default class OpponentSelectionStep extends React.PureComponent<Props> {
                 <th>Wager (Finney)</th>
                 <th>Time initiated</th>
               </tr>
-              {opponents.map(opponent => (
-                <tr key={opponent.address}>
-                  <td>{opponent.name}</td>
-                  <td>{opponent.wager}</td>
+              <tr key={autoOpponentAddress}>
+                <td>RockBot 🤖 </td>
+                <td>30</td>
+                <td>
+                  <Button onClick={() => acceptChallenge(autoOpponentAddress, 30)}>Challenge</Button>
```

Comment:
It's possible to accept a challenge from yourself, which results in a `SEND_MESSAGE` action followed by a `MESSAGE_RECEIVED` action.

![self-selection](https://user-images.githubusercontent.com/8432675/45231304-d42fd100-b288-11e8-8900-f34f88c54b73.gif)

Reply:
I also can't quite figure out how to accept a challenge.

In firefox, I'm signed in using andrew.gord.stewart@gmail.com
In chrome, I'm signed in using andrew@magmo.com

In firefox, I click on the challenge created by the magmo user.
In chrome, I receive a message, but the received message isn't handled.

![propose-fail](https://user-images.githubusercontent.com/8432675/45231558-92ebf100-b289-11e8-9c2b-504b9430db6b.gif)

Reply:
So the challenge stuff needs some work. The idea is that you can only accept challenges from the `waiting-room`, which prevents you accepting your own challenges - hence the hanging when you do. If you leave the `waiting-room` through the app, your challenge will be deleted but if the app crashes or you refresh the page, it remains. To solve these problems, as well as the scenario where the user shuts their laptop, we could use a "heartbeat" approach, where `lastSeen` is updated every 10secs while in the `waiting-room` and the select opponent screen only displays challenges updated in the last 10secs (or so).

Reply:
I think what you're seeing in the gif is a bug. I wasn't able to test this as I could only get one browser working at once :(.

Reply:
Actually, on closer inspection it isn't - to accept the challenge you need to be in the `waiting-room` page - e.g. which you can only get to by creating a new challenge. Try this:

1. andrew@magmo.com creates a new challenge and stays on the waiting page
2. andrew.gord.stewart@gmail.com accepts that challenge

Reply:
so, based on the mechanism we use to accept challenges, challenges should only exist while you're in the "Waiting for someone to accept your challenge" page

the issue is that, if you're on that page, and you then leave the app (eg. you refresh) then the challenge sits there in perpetuity.

The only way around it I've thought is something like:
- firebase stores a `expiresAt` timestamp on a challenge
- when you're in the `waiting-room`, you send a message to firebase every so often, updating the `expiresAt` field to keep the challenge active.

Reply:
Sorry @tomclose for completely ignoring [your comment](https://github.com/magmo/rps-poc/pull/138#discussion_r216020010). It won't happen again.

Reply:
>The idea is that you can only accept challenges from the waiting-room, which prevents you accepting your own challenges

Another edge case to sort out is whether you can be signed into the app from two different locations. This approach wouldn't take care of that edge case.